### PR TITLE
kfctl delete refactor

### PR DESCRIPTION
**Pull request by yanniszark**
Fixes: #293 

Refactor the kfctl deletion algorithm as described in the issue.

### CODE REVIEW

**Comment by yanniszark** [comment]
@adrian555 @Tomcli @kunmingg @jlewi this is ready for review.
I have tested it in my Kubeflow cluster multiple times.

File: pkg/kfapp/kustomize/kustomize.go [hunk 1]
```diff
@@ -350,36 +346,84 @@ func (kustomize *kustomize) Delete(resources kftypesv3.ResourceEnum) error {
 				Message: msg,
 			}
 		}
+	}
```
Comment (adrian555) [inline pkg/kfapp/kustomize/kustomize.go:349]: I wonder if line 339 to line 349 are still needed when line 361 below already handles the empty case.

File: pkg/kfapp/kustomize/kustomize.go [hunk 2]
```diff
@@ -350,36 +346,84 @@ func (kustomize *kustomize) Delete(resources kftypesv3.ResourceEnum) error {
...
+				Message: fmt.Sprintf("error splitting yaml: %v", err),
+			}
+		}
+		for _, r := range resources {
+			err := utils.DeleteResource(r, kubeclient, 5*time.Minute)
+			if err != nil {
 				return &kfapisv3.KfError{
-					Code:    int(kfapisv3.INVALID_ARGUMENT),
-					Message: msg,
+					Code:    int(kfapisv3.INTERNAL_ERROR),
```
Comment (adrian555) [inline pkg/kfapp/kustomize/kustomize.go:417]: Should we continue to delete the rest as a best effort or quit immediately?

Comment (yanniszark) [reply pkg/kfapp/kustomize/kustomize.go:417]: That's a good question :thinking: 
I usually think it's good to fail early.
On the other hand, kubectl tries to delete everything regardless.
We may want to stay close to kubectl behavior as in the future people will probably use it.

File: pkg/kfapp/kustomize/kustomize.go (hunk 1 above)
Comment (yanniszark) [reply pkg/kfapp/kustomize/kustomize.go:349]: Nice catch, you're right!

**Comment by yanniszark** [comment]
@adrian555 I made changes as per the review.
It's ready for another pass whenever possible :smile:

**Comment by adrian555** [comment]
Looks good, thanks @yanniszark.

/lgtm

**Pull request merged**